tests/sles4sap/redirection_tests/ensa2_setup_Netweaver_WebGUI_access.pm

```diff
@@ -0,0 +1,156 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+# Summary: Setup Netweaver WebGUI access
+
+use parent 'sles4sap::sap_deployment_automation_framework::basetest';
+
+package ensa2_setup_Netweaver_WebGUI_access;
+use testapi;
+use sles4sap::console_redirection;
+use sles4sap::console_redirection::redirection_data_tools;
+use sles4sap::sap_host_agent qw(saphostctrl_list_instances);
+use sles4sap::sapcontrol qw(sapcontrol);
+
+=head1 NAME
+
+sles4sap/redirection_tests/ensa2_setup_Netweaver_WebGUI_access.pm.pm - Setup Netweaver WebGUI access on PAS for netweaver instances.
+
+=head1 MAINTAINER
+
+QE-SAP <qe-sap@suse.de>
+
+=head1 DESCRIPTION
+
+Module setup Netweaver WebGUI access for netweaver instances.
+
+B<The key tasks performed by this module include:>
+
+=over
+
+=item * Setup ICM http port on PAS instance via cmdline
+
+=item * Restart NW instance
+
+=item * Configure WebGUI services (via SQL statement)
+
+=item * Activate WebGUI services (via SQL statement)
+
+=item * Restart NW instance again
+
+=item * Check URL is accessible on NW hosts
+
+=back
+
+=head1 OPENQA SETTINGS
+
+Test module does not use any openQA settings
+
+=cut
+
+sub run {
+    my ($self, $run_args) = @_;
+    my $redirection_data = sles4sap::console_redirection::redirection_data_tools->new($run_args->{redirection_data});
+
+    # Get PAS host
+    my %pas_host = %{$redirection_data->get_pas_host};
+    if (!%pas_host) {
+        record_info('N/A', 'Netweaver deployment not detected, skipping.');
+        return;
+    }
+
+    # Everything is now executed on SUT, not worker VM
+    my $host = join('', keys(%pas_host));
```

Comment:
this might need to be sanitized. 
In case there is for some reason more than one PAS instance in data. 
You would get two joined hostnames. 
You can maybe just let test fail if there are more hosts than one.